micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -1282,9 +1285,62 @@ public String getContextualName() {
                     return contextualName;
                 }
 
+                @Override
+                public long getTimestamp() {
+                    return this.timestamp;
+                }
+
+                @Override
+                public TimeUnit getTimestampTimeUnit() {
+                    return TimeUnit.MILLISECONDS;
```

Comment:
Is this needed? Can we just say in the javadocs that it is the current wall time in ms?

Reply:
That's what I did originally. I even named the method `getTimestampInMs` or sth like that. With 2 methods there is no doubt about what the units are and you are able to modify it as you please. Without that you can make easy mistakes (like I did when I've forgotten that timestamp in Brave is micro not milliseconds :shrug: ). WDYT @shakuzen ?